### PR TITLE
Enable proper storing, loading and display of doctors

### PR DESCRIPTION
**Pull request by praveenkrishna0512**
Current situation before changes
- The `Storage` and `UI` components were referring to the previously used `filteredPersonsList`, which is no longer in use.

What's been done
- The `Storage` and `UI` components have been updated to use the `filteredDoctorList` to reference all doctors.

### CODE REVIEW

**Review by owen-yap — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/ui/DoctorCard.java [hunk 1]
```diff
@@ -40,16 +40,16 @@ public class PersonCard extends UiPart<Region> {
...
-        this.person = person;
+        this.doctor = doctor;
         id.setText(displayedIndex + ". ");
-        name.setText(person.getName().fullName);
-        phone.setText(person.getPhone().value);
-        email.setText(person.getEmail().value);
-        person.getTags().stream()
+        name.setText(doctor.getName().fullName);
+        phone.setText(doctor.getPhone().value);
+        email.setText(doctor.getEmail().value);
```
[the remark this answers is not in the document]
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/ui/DoctorCard.java:51]: These will be shown in the enlarged contact card, not the normal one

File: src/main/java/seedu/address/ui/DoctorCard.java [hunk 2]
```diff
@@ -7,14 +7,14 @@
 import javafx.scene.layout.FlowPane;
 import javafx.scene.layout.HBox;
 import javafx.scene.layout.Region;
-import seedu.address.model.person.Person;
+import seedu.address.model.person.Doctor;
 
 /**
- * An UI component that displays the name, phone number, email and tags of a {@code Person}.
+ * An UI component that displays the name, phone number, email and tags of a {@code Doctor}.
```
[the remark this answers is not in the document]
Comment (praveenkrishna0512) [reply src/main/java/seedu/address/ui/DoctorCard.java:13]: These will be shown in the enlarged contact card, not the normal one

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 3]
```diff
@@ -63,12 +58,12 @@ private static JsonAdaptedPerson convertToJsonAdaptedPerson(Person person) {
      */
     public AddressBook toModelType() throws IllegalValueException {
         AddressBook addressBook = new AddressBook();
-        for (JsonAdaptedPerson jsonAdaptedPerson : persons) {
-            Person person = jsonAdaptedPerson.toModelType();
-            if (addressBook.hasPerson(person)) {
+        for (JsonAdaptedDoctor jsonAdaptedDoctor : doctors) {
+            Doctor doctor = jsonAdaptedDoctor.toModelType();
+            if (addressBook.hasDoctor(doctor)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_PERSON);
```
Comment (Bobfree546) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:64]: Replace with `MESSAGE_DUPLICATE_DOCTOR`

File: src/main/java/seedu/address/ui/DoctorCard.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (owen-yap) [reply src/main/java/seedu/address/ui/DoctorCard.java:51]: Okie

File: src/main/java/seedu/address/storage/JsonSerializableAddressBook.java [hunk 4]
```diff
@@ -12,24 +12,23 @@
...
-import seedu.address.model.person.Person;
 
 /**
  * An Immutable AddressBook that is serializable to JSON format.
  */
 @JsonRootName(value = "docedex")
 class JsonSerializableAddressBook {
 
-    public static final String MESSAGE_DUPLICATE_PERSON = "Persons list contains duplicate person(s).";
+    public static final String MESSAGE_DUPLICATE_PERSON = "Doctors list contains duplicate doctor(s).";
```
Comment (praveenkrishna0512) [inline src/main/java/seedu/address/storage/JsonSerializableAddressBook.java:22]:
```suggestion
    public static final String MESSAGE_DUPLICATE_DOCTOR = "Doctors list contains duplicate doctor(s).";
```

**Review by vigonometry — APPROVED** [review]
LGTM

**Review by owen-yap — APPROVED** [review]
LGTM

**Pull request merged**